app/lib/submission_builder/ty2024/states/md/md_return_xml.rb

```diff
@@ -29,12 +29,20 @@ def schema_file
           end
 
           def supported_documents
+            calculated_fields = @submission.data_source.tax_calculator.calculate
+            has_income_from_taxable_pensions_iras_annuities = calculated_fields.fetch(:MD502_LINE_1D)&.to_i.positive?
+
             supported_docs = [
               {
                 xml: SubmissionBuilder::Ty2024::States::Md::Documents::Md502,
                 pdf: PdfFiller::Md502Pdf,
                 include: true
               },
+              {
+                xml: SubmissionBuilder::Ty2024::States::Md::Documents::Md502R,
+                pdf: PdfFiller::Md502RPdf,
+                include: has_income_from_taxable_pensions_iras_annuities
```

Comment:
can we remove lines 32 and 33 and instead just do
```suggestion
                include:  @submission.data_source.direct_file_data.fed_taxable_pensions&.to_i.positive?
```

Reply:
I would prefer to access the calculator, since it lets us avoid a very small amount of code duplication. I was following the example that Jenny shared with me of [`ny_return_xml.rb`](https://github.com/codeforamerica/vita-min/blob/main/app/lib/submission_builder/ty2022/states/ny/ny_return_xml.rb#L135) where the calculator is used to determine conditions for including various documents